### PR TITLE
upgrading depenencies to also work with Sphinx 4

### LINKED ISSUE #10: Version 0.0.4 not compatible with latest version of Sphinx (4.3.2)
Hey,
trying to upgrade to the latest version of Sphinx - version 4.3.2 - but this module doesn't seem to be compatible.
Any plans on addressing this, or any clues on how to fix this?

When running `pip3 install -U Sphinx`, I'm getting the below error:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
sphinx-charts 0.0.4 requires sphinx<4,>=2, but you have sphinx 4.3.2 which is incompatible.
```

And when trying to build the site anyway, then this error happens:
```Exception occurred:
  File "/usr/local/lib/python3.9/site-packages/sphinx_charts/charts.py", line 242, in setup
    app.add_javascript(path)
AttributeError: 'Sphinx' object has no attribute 'add_javascript'
The full traceback has been saved in /var/folders/p8/h425k82s2r3cy03hyqyhst7w0000gp/T/sphinx-err-sxpazmrg.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Command exited with exit code: 2
The server will continue serving the build folder, but the contents being served are no longer in sync with the documentation sources. Please fix the cause of the error above or press Ctrl+C to stop the server.
```

### PR DESCRIPTION
**Pull request by LydByDissing**
Fixes #10.

* upgrading to Sphinx 4

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{36,37,38}-sphinx{2,3},py36-flake8,docs
+envlist = py{36,37,38,39}-sphinx{2,3,4},py{36,37,38,39}-flake8,docs
```
Comment (thclark) [inline tox.ini:2]: There's no need to have flake8 run 4 times, stick to the one variant:
```
py{36,37,38,39}-sphinx{2,3,4},py36-flake8,docs
```

File: sphinx_charts/charts.py [hunk 2]
```diff
@@ -239,7 +239,7 @@ def setup(app):
             if "add_script_file" in dir(app):
                 app.add_script_file(path)
             else:
-                app.add_javascript(path)
+                app.add_js_file(path)
```
Comment (thclark) [inline sphinx_charts/charts.py:242]: To maintain compatibility with Sphinx 2, while preferably using the latest sphinx API on the first attempt, you'd do:
```py
try:
    app.add_js_file(path)
except AttributeError:
    # For backward compatibility with Sphinx 2 and 3
    app.add_javascript(path)
```

File: setup.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 setup(
     name="sphinx_charts",
-    version="0.0.4",
+    version="0.1",
```
Comment (thclark) [inline setup.py:19]: Use 0.1.0 as a version so the release gets tagged correctly on pypi and GitHub

**Review by thclark — CHANGES_REQUESTED** [review]
Nice work, Tue!

A couple of tweaks and this'll be production-ready, then I'll merge and make the release. I'll also update the release automation while I'm at it, make it easier for next time.

Thanks for your contributions! :)

File: tox.ini (hunk 1 above)
Comment (LydByDissing) [reply tox.ini:2]: ah, makes sense. Haven't used `tox`before,

**Comment by LydByDissing** [comment]
Hopefully this last commit is a little better. Needed to upgrade additional parts to make it work. 
Also not sure that the tests caught the bug in the first commit...so maybe good to validate against Sphinx 2 and 3.

**Comment by LydByDissing** [comment]
Hi Tom,

No stress - I can work.
So do it in your time :-)

With regards

Tue Dissing
CEO and founder
Lyd by Dissing

+45 20 68 33 02
***@***.***

**Comment by thclark** [comment]
Woops closed accidentally triggered by me switching from 'master' to 'main'. Sorry @LydByDissing - I'll resurrect this and am just getting it merged now.

**Pull request merged**